test/methods.js

```diff
@@ -294,6 +294,166 @@ asyncTest("remote", function() {
 	strictEqual( v.element(e), true, "still invalid, because remote validation must block until it returns; dependency-mismatch considered as valid though" );
 });
 
+asyncTest("remote success", function() {
+	expect(3);
+
+	var e = $("#username"),
+		v = $("#userForm").validate({
+			rules: {
+				username: {
+					required: true,
+					remote: "valid.php"
+				}
+			},
+			messages: {
+				username: {
+					required: "Please",
+					remote: jQuery.validator.format("{0} in use")
+				}
+			},
+			submitHandler: function() {
+				ok( false, "submitHandler may never be called when validating only elements");
+			}
+		});
+
+	$(document).ajaxStop(function() {
+		$(document).unbind("ajaxStop");
+		strictEqual( v.element(e), true );
+		start();
+	});
+	strictEqual( v.element(e), false);
+	e.val("Peter");
+	strictEqual( v.element(e), true);
+});
+
+asyncTest("remote failure", function() {
+	expect(3);
+
+	var e = $("#username"),
```

Comment:
Maybe you could use more explicit variable names than the one-letter-long ones?

Reply:
It looks like that's convention used in those tests, if you'll take a look at the other tests all elements are called `e` and validators are called `v`

Reply:
That's a shitty convention :P but ok, let's not overcomplicate this review and keep things consistent.

Reply:
Noone from us chose that - it's a fork of https://github.com/jquery-validation/jquery-validation/ XD